lua/lspconfig/leanls.lua

```diff
@@ -1,23 +1,36 @@
 local configs = require 'lspconfig/configs'
 local util = require 'lspconfig/util'
 
+local function find_toml(startpath)
+  return util.search_ancestors(util.path.dirname(startpath), function(path)
+    local toml_file = util.path.join(path, "leanpkg.toml")
+    if util.path.is_file(toml_file) then
+      return path
+    end
+  end)
+end
```

Comment:
This function looks a bit redundant to me. I would suggest you just use `util.root_pattern` instead (see comment below). It's what most configs use.

Reply:
Agreed. Thank you!